core/templates/dev/head/domain/collection/CollectionUpdateService.js

```
@@ -257,21 +279,30 @@ oppia.factory('CollectionUpdateService', [
         },
 
         /**
+         * Returns whether the collection tag is valid.
+         */
+        isTagValid: function(collection, tags) {
+          return _validateTags(tags);
+        },
+
+        /**
          * Changes the tags of a collection and records the change in
          * the undo/redo service.
          */
         setCollectionTags: function(collection, tags) {
           var oldTags = angular.copy(collection.getTags());
-          _applyPropertyChange(
-            collection, COLLECTION_PROPERTY_TAGS, tags, oldTags,
-            function(changeDict, collection) {
-              // Apply.
-              var tags = _getNewPropertyValueFromChangeDict(changeDict);
-              collection.setTags(tags);
-            }, function(changeDict, collection) {
-              // Undo.
-              collection.setTags(oldTags);
-            });
+          if (_validateTags(tags)) {
```

This should throw an error if tags are not valid, instead of failing silently.